numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
I don't need to here, because `check_and_adjust_axis_msg` takes a borrowed reference. It doesn't actually need a refcount at all until it makes it to the python code, and `PyObject_CallFunction()` increments the refcount for me, AFAICT.